### ISSUE TITLE
PEAR.WhiteSpace.ScopeIndent.IncorrectExact conflict with PEAR.Functions.FunctionCallSignature.Indent

### ISSUE THREAD

**Issue by alasdairdc** [squizlabs/PHP_CodeSniffer#528]
Using the test file and PEAR.WhiteSpace.ScopeIndent exact true:
```php
<?php
function test()
{
    echo 'test';
    echo 'test2';
     echo 'test3';
     if (true) {
     echo 'test3';
     }  
     echo 'test3';
     $x = f1(
     'test1', 'test2',
     'test3'
     ); 
}
```
and running against `--standard=PEAR` reports the following:
```
-----------------------------------------------------------------------------------------------------------------------------------------------
FOUND 10 ERRORS AFFECTING 8 LINES
-----------------------------------------------------------------------------------------------------------------------------------------------
  6 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 5 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
  7 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 5 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
  8 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found 5 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
  9 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found 5 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
 10 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found 5 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
 11 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found 5 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
 12 | ERROR | [x] Multi-line function call not indented correctly; expected 9 spaces but found 5 (PEAR.Functions.FunctionCallSignature.Indent)
 12 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found 5 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
 13 | ERROR | [x] Multi-line function call not indented correctly; expected 9 spaces but found 5 (PEAR.Functions.FunctionCallSignature.Indent)
 13 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found 5 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
-----------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 10 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------------------------------
```

the resultant fix is:
```php
<?php
function test()
{
    echo 'test';
    echo 'test2';
    echo 'test3';
    if (true) {
        echo 'test3';
    }   
    echo 'test3';
    $x = f1( 
        'test1', 'test2',
        'test3'
    );  
}
```
as expected however running CS again results in
```
---------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
---------------------------------------------------------------------------------------------------------------------
 12 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 8 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
 13 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 8 (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
---------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
---------------------------------------------------------------------------------------------------------------------
```
fixing again:
```php
<?php
function test()
{
    echo 'test';
    echo 'test2';
    echo 'test3';
    if (true) {
        echo 'test3';
    }   
    echo 'test3';
    $x = f1( 
    'test1', 'test2',
    'test3'
    );  
}
```
and testing:
```
-----------------------------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
-----------------------------------------------------------------------------------------------------------------------------------------------
 12 | ERROR | [x] Multi-line function call not indented correctly; expected 8 spaces but found 4 (PEAR.Functions.FunctionCallSignature.Indent)
 13 | ERROR | [x] Multi-line function call not indented correctly; expected 8 spaces but found 4 (PEAR.Functions.FunctionCallSignature.Indent)
-----------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------------------------------
```
is now cyclical between the last two outputs.

Can we get these two to play nicely together when exact true is marked to fix the indentation for non functions?

Was planning on using this in a pre-commit hook to fix/report sloppy code, including extra indentation hence the exact.

**Comment by gsherwood** [comment]
The only thing I could do is to turn off the exact flag when checking content within parenthesis, such as inside function calls and IF statement conditions.

This should do what you want, but you may find other parts of code where the exact flag has been turned off and you don't have a specific sniff to check for rules in that area. If you find code like that, you'll need to find/write a sniff to do the check for you.
